pkg/storage/replica_state.go

```diff
@@ -210,6 +210,47 @@ func setAppliedIndex(
 		nil /* txn */)
 }
 
+func setAppliedIndexBlind(
+	ctx context.Context,
+	eng engine.ReadWriter,
+	ms *enginepb.MVCCStats,
+	rangeID roachpb.RangeID,
+	appliedIndex,
+	leaseAppliedIndex uint64,
+) error {
+	var value roachpb.Value
+	value.SetInt(int64(appliedIndex))
+
+	if err := engine.MVCCBlindPut(ctx, eng, ms,
+		keys.RaftAppliedIndexKey(rangeID),
+		hlc.ZeroTimestamp,
+		value,
+		nil /* txn */); err != nil {
+		return err
+	}
+	value.SetInt(int64(leaseAppliedIndex))
+	return engine.MVCCBlindPut(ctx, eng, ms,
+		keys.LeaseAppliedIndexKey(rangeID),
+		hlc.ZeroTimestamp,
+		value,
+		nil /* txn */)
+}
+
+func calcAppliedIndexSysBytes(
```

Comment:
Let me see about refactoring this so that we don't call `setAppliedIndexBlind`. It should be feasible to avoid that.